src/defaults/runningTerminals.tsx

```diff
@@ -0,0 +1,108 @@
+import * as ReactDOM from 'react-dom';
+import * as React from 'react';
+
+import {
+    JupyterLabPlugin,
+    JupyterLab,
+    ApplicationShell
+} from '@jupyterlab/application';
+import { IDefaultStatusesManager } from './manager';
+
+import { TerminalManager, TerminalSession } from '@jupyterlab/services';
+
+import { Widget } from '@phosphor/widgets';
+
+export namespace RunningTerminalComponent {
+    export interface IState {
+        terminalSession: number;
+    }
+    export interface IProps {
+        terminalManager: TerminalManager;
+        shellHost: ApplicationShell;
+    }
+}
+
+export class RunningTerminalComponent extends React.Component<
+    RunningTerminalComponent.IProps,
+    RunningTerminalComponent.IState
+> {
+    state = {
+        terminalSession: 0
+    };
+
+    constructor(props: RunningTerminalComponent.IProps) {
+        super(props);
+        this.props.terminalManager.runningChanged.connect(this.updateTerminal);
+    }
+
+    componentDidMount() {
+        TerminalSession.listRunning().then(value =>
+            this.setState({ terminalSession: value.length })
+        );
+    }
+
+    updateTerminal = (
+        terminalManage: TerminalManager,
+        terminals: TerminalSession.IModel[]
+    ) => {
+        this.setState({ terminalSession: terminals.length });
+    };
+
+    handleClick = () => {
+        this.props.shellHost.expandLeft();
+        this.props.shellHost.activateById('jp-running-sessions');
+    };
+    render() {
+        return (
+            <div onClick={this.handleClick}>
+                {this.state.terminalSession} Terminals Active
+            </div>
+        );
+    }
+}
+
+export class RunningTerminals extends Widget {
+    constructor(opts: RunningTerminals.IOptions) {
+        super();
+        this._manager = new TerminalManager();
```

Comment:
Why are you creating a new `TerminalManager` here? Is it possible to get that from somewhere already created?